route/v1/types.go

```diff
@@ -284,6 +285,26 @@ type TLSConfig struct {
 	//
 	// +kubebuilder:validation:Enum=Allow;None;Redirect;""
 	InsecureEdgeTerminationPolicy InsecureEdgeTerminationPolicyType `json:"insecureEdgeTerminationPolicy,omitempty" protobuf:"bytes,6,opt,name=insecureEdgeTerminationPolicy,casttype=InsecureEdgeTerminationPolicyType"`
+
+	// externalCertificate provides certificate contents as a secret reference.
+	// This should be a single serving certificate, not a certificate
+	// chain. Do not include a CA certificate. The secret referenced should
+	// be present in the same namespace as that of the Route. If externalCertificate is
+	// provided, do not set certificate.
```

Comment:
I would make this a stronger requirement here
```suggestion
	// be present in the same namespace as that of the Route.
	// Forbidden when `certificate` is set.
```

Reply:
Addressed